server/src/main/java/org/cloudfoundry/identity/uaa/client/VmidentityQueryableClientDetailsService.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ *     Cloud Foundry
+ *     Copyright (c) [2017] VMware, Inc. All Rights Reserved.
+ *
+ *     This product is licensed to you under the Apache License, Version 2.0 (the "License").
+ *     You may not use this product except in compliance with the License.
+ *
+ *     This product includes a number of subcomponents with
+ *     separate copyright notices and license terms. Your use of these
+ *     subcomponents is subject to the terms and conditions of the
+ *     subcomponent's license, as noted in the LICENSE file.
+ *******************************************************************************/
+package org.cloudfoundry.identity.uaa.client;
+
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cloudfoundry.identity.uaa.resources.Queryable;
+import org.cloudfoundry.identity.uaa.resources.QueryableResourceManager;
+import org.cloudfoundry.identity.uaa.zone.MultitenantVmidentityClientDetailsService;
+import org.springframework.security.oauth2.provider.ClientDetails;
+
+public class VmidentityQueryableClientDetailsService implements Queryable<ClientDetails>,
```

Comment:
It's mostly for the query call. There's a JDBC variant `JdbcQueryableClientDetailsService` that does basically the same thing we're doing here, but in the bean configuration, requires both a `JdbcTemplate` and `JdbcPagingListFactory`, neither of which do we have any use for.